### PR TITLE
Log tracking method used and required installs for tracking CPU

### PR DESCRIPTION
**Pull request by rosekelly6400**
Addresses Issue #585 

- moved the code that sets the ram, cpu and gpu tracking into its own function
- added additional logging saying what RAM, CPU and GPU tracking methods are being used and logging giving instructions on what needs to be installed to improve tracking accuracy

### CODE REVIEW

**Comment by rosekelly6400** [comment]
@benoit-cty @inimaz this pull request is for addressing issue [#585: Log steps to install needed codecarbon dependencies ](https://github.com/mlco2/codecarbon/issues/585) Is this what you were looking for? Happy to make more changes if needed

File: codecarbon/emissions_tracker.py [hunk 1]
```diff
@@ -338,16 +387,31 @@ def __init__(
             self._conf["gpu_model"] = hardware_gpu.get_model()
             self._conf["gpu_count"] = 1
         else:
+            # PRINT WHAT TO INSTALL TO INCREASE ACCURACY
+            #if darwin
+            cpu_tracking_install_instructions = ""
+            if is_mac_os():
+                if "M1" in detect_cpu_model() or "M2" in detect_cpu_model() or "M3" in detect_cpu_model():
+                    cpu_tracking_install_instructions = ""
+                cpu_tracking_install_instructions = "Mac OS and ARM processor detected: Please enable PowerMetrics sudo to measure CPU"
```
Comment (benoit-cty) [inline codecarbon/emissions_tracker.py:396]:
```suggestion
                    cpu_tracking_install_instructions = "Mac OS and ARM processor detected: Please enable PowerMetrics sudo to measure CPU"
```

File: codecarbon/emissions_tracker.py [hunk 2]
```diff
@@ -338,16 +387,31 @@ def __init__(
             self._conf["gpu_model"] = hardware_gpu.get_model()
             self._conf["gpu_count"] = 1
         else:
+            # PRINT WHAT TO INSTALL TO INCREASE ACCURACY
```
Comment (benoit-cty) [inline codecarbon/emissions_tracker.py:390]:
```suggestion
            # Explain what to install to increase accuracy
```

**Review by benoit-cty — CHANGES_REQUESTED** [review]
Thanks a lot !
You need to add `import sys` to `util.py`.

**Comment by rosekelly6400** [comment]
> Thanks a lot ! You need to add `import sys` to `util.py`.

@benoit-cty all the requested changes have been added and pushed

**Comment by rosekelly6400** [comment]
@benoit-cty sorry was having some issues running the hatch scripts locally to test the pipelines but got them working now, hopefully the pipeline should pass now

**Review by benoit-cty — APPROVED** [review]

**Comment by benoit-cty** [comment]
Thanks, your contribution will be in the 2.7.0 that I will release today.

**Pull request merged**